src/databricks/labs/remorph/reconcile/query_builder/hash_query.py

```diff
@@ -0,0 +1,58 @@
+import sqlglot.expressions as exp
+
+from databricks.labs.remorph.reconcile.query_builder.base import QueryBuilder
+from databricks.labs.remorph.reconcile.query_builder.expression_generator import (
+    build_column,
+    concat,
+    get_hash_transform,
+    lower,
+    transform_expression,
+)
+from databricks.labs.remorph.reconcile.recon_config import Schema
+
+
+class HashQueryBuilder(QueryBuilder):
+    def build_query(self) -> str:
+        hash_cols = sorted((self.join_columns | self.select_columns) - self.threshold_columns - self.drop_columns)
+        key_cols = sorted(self.join_columns | self.partition_column)
+
+        cols_with_alias = [
+            build_column(this=col, alias=self.table_conf.get_tgt_to_src_col_mapping(col, self.layer))
+            for col in key_cols
+        ]
+
+        key_cols_with_transform = self._add_transformations(cols_with_alias, self.source)
+        hash_col_with_transform = [self._generate_hash_algorithm(hash_cols)]
+
+        res = (
+            exp.select(*hash_col_with_transform + key_cols_with_transform)
+            .from_(":tbl")
+            .where(self.filter)
+            .sql(dialect=self.source)
+        )
+
+        return res
+
+    def _add_transformations(self, aliases: list[exp.Alias], source: str) -> list[exp.Alias]:
```

Comment:
This method looks generic enough to move to `QueryBuilder` class, thoughts?

Reply:
Done.